txpipe/map_tests.py

```diff
@@ -0,0 +1,162 @@
+from .base_stage import PipelineStage
+from .data_types import MapsFile, FileCollection
+import pathlib
+import glob
+import numpy as np
+
+class TXMapCorrelations(PipelineStage):
+    name = "TXMapCorrelations"
+    inputs = [
+        ("lens_maps", MapsFile),
+        ("convergence_maps", MapsFile),
+        ("source_maps", MapsFile),
+        ("mask", MapsFile),
+    ]
+
+    outputs = [
+        ("map_systematic_correlations", FileCollection),
+    ]
+
+    config_options = {
+        "supreme_path_root": "/global/cscratch1/sd/erykoff/dc2_dr6/supreme/supreme_dc2_dr6d_v2",
+        "nbin": 20,
+        "outlier_fraction": 0.05,
+    }
+
+    def read_healsparse(self, map_path, nside):
+        import healsparse
+        import healpy
+
+        # Convert to correct res healpix map
+        m = healsparse.HealSparseMap.read(map_path)
+        m = m.generate_healpix_map(nside=nside)
+        m = healpy.ud_grade(m, nside, order_in="nest", order_out="ring")
```

Comment:
I think the recommendation is not to use healpy's `ud_grade` function as it's not doing things 100% correctly. Better to do map2alm and alm2map (setting the nside here)

Reply:
I didn't know that - interesting!  Although for this case the nside for the map will not change, so it's just re-indexing the pixels from ring to nest (because we generated the map at the correct nside in the first place), which I assume should be fine?  Having said that, we are using healsparse's generate_healpix map to go from the input to the output nside, and I have no idea how that works - will check.

Reply:
By default healsparse just takes the mean of hit pixels in the high-res map, which I think is the right thing to do here.

Reply:
I've added a comment noting that the resolution doesn't change.